### PR TITLE
Add darwin support for remote-client

### PR DESCRIPTION
**Pull request by baude**
Add the ability to cross-compile podman remote for OSX.

Also, add image exists and tag to remote-client.

Signed-off-by: baude <bbaude@redhat.com>

### CODE REVIEW

File: pkg/rootless/rootless_unsupported.go [hunk 1]
```diff
@@ -14,7 +15,8 @@ func IsRootless() bool {
 // BecomeRootInUserNS is a stub function that always returns false and an
 // error on unsupported OS's
 func BecomeRootInUserNS() (bool, int, error) {
-	return false, -1, errors.New("this function is not supported on this os")
+	debug.PrintStack()
+	return false, -1, errors.New("this function is not supported on this os1")
```
Comment (mheon) [inline pkg/rootless/rootless_unsupported.go]: Debug stuff slipped in?

File: libpod/container.go [hunk 2]
```diff
@@ -365,6 +366,25 @@ type ContainerConfig struct {
 	Systemd bool `json:"systemd"`
 }
 
+// AttachStreams contains streams that will be attached to the container
```
Comment (mheon) [inline libpod/container.go]: Throw this in container_api.go right above the Attach() function

File: libpod/container_attach.go [hunk 3]
```diff
@@ -1,3 +1,5 @@
+//+build linux
```
Comment (mheon) [inline libpod/container_attach.go]: Rename file to container_attach_linux

File: cmd/podman/podman.go [hunk 4]
```diff
@@ -0,0 +1,16 @@
+// +build linux
```
Comment (mheon) [inline cmd/podman/podman.go]: Can we get a better name on this file? registries_linux.go and registries_unsupported.go maybe?

File: cmd/podman/podman.go [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+package main
+
+import (
+	"github.com/sirupsen/logrus"
+	"os"
```
Comment (mheon) [inline cmd/podman/podman.go]: Import ordering

File: cmd/podman/podman.go [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+)
+
+func CheckForRegistries() {
+	if _, err := os.Stat("/etc/containers/registries.conf"); err != nil {
+		if os.IsNotExist(err) {
+			logrus.Warn("unable to find /etc/containers/registries.conf. some podman (image shortnames) commands may be limited")
```
Comment (mheon) [inline cmd/podman/podman.go]: What about this is linux-specific? This seems like it ought to build on OS X fine

File: cmd/podman/main.go [hunk 7]
```diff
@@ -120,7 +120,7 @@ func main() {
 			os.Exit(1)
 		}
 		args := c.Args()
-		if args.Present() {
+		if args.Present() && rootless.IsRootless() {
```
Comment (mheon) [inline cmd/podman/main.go]: @giuseppe Is this safe (is rootless.IsRootless() set appropriately at this point?)

Comment (baude) [reply cmd/podman/main.go]: hes the one who advised to do this :)  but lets make sure

File: cmd/podman/podman.go (hunk 6 above)
Comment (baude) [reply cmd/podman/podman.go]: there is no use for registries.conf on the remote client side.  all of that is handled on the server side.

File: cmd/podman/podman.go (hunk 4 above)
Comment (baude) [reply cmd/podman/podman.go]: i needed a generic file for stuff like this.  there will be more which is why i chose it.  how about platform.go ?

Comment (mheon) [reply cmd/podman/podman.go]: Platform.go SGTM

File: cmd/podman/exists.go [hunk 8]
```diff
@@ -1,6 +1,7 @@
 package main
 
 import (
+	"github.com/containers/libpod/libpod/adapter"
```
Comment (TomSweeneyRedHat) [inline cmd/podman/exists.go]: gofmt ok with this here instead of below near line 7?

File: libpod/adapter/client.go [hunk 9]
```diff
@@ -8,7 +8,8 @@ import (
 
 // Connect provides a varlink connection
 func (r RemoteRuntime) Connect() (*varlink.Connection, error) {
-	connection, err := varlink.NewConnection("unix:/run/podman/io.podman")
+	//connection, err := varlink.NewConnection("unix:/run/podman/io.podman")
```
Comment (TomSweeneyRedHat) [inline libpod/adapter/client.go]: might want to clear this line at some point

File: pkg/rootless/rootless_unsupported.go (hunk 1 above)
Comment (baude) [reply pkg/rootless/rootless_unsupported.go]: fixed

File: libpod/container.go (hunk 2 above)
Comment (baude) [reply libpod/container.go]: done

File: libpod/container_attach.go (hunk 3 above)
Comment (baude) [reply libpod/container_attach.go]: done

File: cmd/podman/podman.go (hunk 4 above)
Comment (baude) [reply cmd/podman/podman.go]: done and done

File: cmd/podman/podman.go (hunk 5 above)
Comment (baude) [reply cmd/podman/podman.go]: done

File: libpod/adapter/client.go (hunk 9 above)
Comment (baude) [reply libpod/adapter/client.go]: yes but worth tidying up, ty

File: cmd/podman/exists.go (hunk 8 above)
Comment (baude) [reply cmd/podman/exists.go]: yes but worth tidying up, ty

**Comment by baude** [comment]
merge me please?

**Comment by rhatdan** [comment]
LGTM

File: cmd/podman/podman.go (hunk 6 above)
Comment (mheon) [reply cmd/podman/podman.go]: Swapping to a LocalRuntime method does sound like a better approach

**Pull request merged**